### ISSUE TITLE
Fix a11y issue on heading permalink icons

### ISSUE THREAD

**Issue by igorkorenfeld** [18F/guides#104]
### A description of the work
In reviewing the work on #341 @jasnakai [noted](https://github.com/18F/guides/pull/88#pullrequestreview-1702287428) that the way the permalink icons are implemented on headings in the replatformed guides (i.e. the link icon next to headings) is not properly structured. Namely, the `<a>` element is in a `span` element that have an `aria-hidden=true` attribute, but the anchor is focusable. We should edit the current implementation so that it is properly structured for accessibility. Note that the implementation on the guides is borrowed from the TTS handbook, so any finding here should be applied to the handbook as well (though not necessarily as part of work on this issue). 

Our current implementation uses the `markdown-it-anchor` [plugin](https://www.npmjs.com/package/markdown-it-anchor#link-after-header), which has been updated with more accessible options. It would be good to see if any of the recent updates may be useful to us.

### Point of contact on this issue
Who can we follow-up with if we have questions?
@jasnakai 

### Reproduction steps (if necessary)
Be as specific as possible

**Billable?**
- [ ] Yes
- [X] No

**Skills needed**
- [ ] Any human
- [ ] Design
- [ ] Content
- [X] Engineering
- [ ] Acquisition
- [ ] Product
- [ ] Other

**Timeline**
Does this need to happen in the next two weeks?
- [ ] Yes
- [X] No

### How much time do you anticipate this work taking?
Medium size

### Acceptance Criteria
- [ ] The permalink icons are structured in an accessible way.

**Comment by jasnakai** [comment]
I reviewed the update to the markdown-it-anchor plugin and it will work. Moving forward with implementation.

**Comment by jasnakai** [comment]
Needs further review of the custom code currently in use.

**Comment by jasnakai** [comment]
Updating the plugin did not completely fix the issue. Updating the related JS (config/headingLinks.js) fixed the issue.
